code/src/App.js

```diff
@@ -1,9 +1,69 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+
+import ThoughtList from 'components/ThoughtList';
+import ThoughtForm from 'components/ThoughtForm';
 
 export const App = () => {
+  const [thoughtList, setThoughtList] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [newThought, setNewThought] = useState('');
+
+  const fetchThoughts = () => {
+    setLoading(true);
+    fetch('https://happy-thoughts-technigo.herokuapp.com/thoughts')
+      .then((res) => res.json())
+      .then((data) => setThoughtList(data))
+      .catch((error) => console.error(error))
+      .finally(() => setLoading(false));
+  }
+
+  useEffect(() => {
+    fetchThoughts();
+  }, []);
+
+  const handleNewLikeChange = (_id) => {
+    const options = { method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      } }
+    fetch(`https://happy-thoughts-technigo.herokuapp.com/thoughts/${_id}/like`, options)
+      .then((res) => res.json())
+      .then(() => fetchThoughts())
+  }
+
+  const handleNewThoughtChange = (event) => {
+    setNewThought(event.target.value)
+  }
+
+  const onFormSubmit = (event) => {
+    event.preventDefault();
+
+    const options = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        message: newThought
+      })
+    }
+
+    fetch('https://happy-thoughts-technigo.herokuapp.com/thoughts', options)
+      .then((res) => res.json())
+      .then(() => fetchThoughts())
+      .finally(() => setNewThought(''));
+  }
```

Comment:
Great job with all the fetches and post, this was tricky to create! But in the future, since this is a very long and a little complicated const, maybe think about the order of every part in it so it's easier to follow. And comments what the different part does. I didn't comment this in my code either but now when I read yours I realise it's needed 😅